### ISSUE TITLE
Connecting to Puppet DB over HTTPS (TLS)

### ISSUE THREAD

**Issue by dannyk81** [camptocamp/prometheus-puppetdb#3]
Hi, first off - thanks for building this tool 👍 

Our Puppet DB is exposed via an HTTPS endpoint and requires client certificate authentication.

Is there any way/plan to support this? as right now it doesn't seem possible.

**Comment by dannyk81** [comment]
I was able to do a quick hack to make it work, it's a bit dirty but seems to be pointing to the right direction.

```
diff --git a/main.go b/main.go
index d63c2f9..bee66be 100644
--- a/main.go
+++ b/main.go
@@ -6,6 +6,7 @@ import (
        "fmt"
        "io/ioutil"
        "net/http"
+       "crypto/tls"
        "os"
        "strings"
        "time"
@@ -15,7 +16,7 @@ import (
        "gopkg.in/yaml.v2"
 )

 type Config struct {
        Version     bool   `short:"V" long:"version" description:"Display version."`
@@ -65,7 +66,17 @@ func main() {
                os.Exit(1)
        }

-       client := &http.Client{}
+       cert, err := tls.LoadX509KeyPair("certs/client.pem", "certs/client.key")
+       if err != nil {
+               fmt.Println(err)
+               os.Exit(1)
+       }
+
+       tr := &http.Transport{
+               TLSClientConfig: &tls.Config{Certificates: []tls.Certificate{cert}, InsecureSkipVerify: true},
+       }
+
+       client := &http.Client{Transport: tr}

        for {
                nodes, err := getNodes(client, cfg.PuppetDBURL, cfg.Query)
```

/edit: "certs/client.pem", "certs/client.key" are the client's cert and key...

**Comment by raphink** [comment]
Fixed in #4
